--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -327,13 +327,20 @@
 	set( LIBS tester lapacktest magma )  ##magmaf )
 else()
 	# Unix doesn't seem to have a problem with mixing C, CUDA, and Fortran files
-	cuda_add_library( magma ${libmagma_all} )
-	target_link_libraries( magma
-		${LAPACK_LIBRARIES}
-		${CUDA_CUDART_LIBRARY}
-		${CUDA_CUBLAS_LIBRARIES}
-		${CUDA_cusparse_LIBRARY}
-	)
+	if ( CUDA_FOUND )
+		cuda_add_library( magma ${libmagma_all} )
+		target_link_libraries( magma
+			${LAPACK_LIBRARIES}
+			${CUDA_CUDART_LIBRARY}
+			${CUDA_CUBLAS_LIBRARIES}
+			${CUDA_cusparse_LIBRARY}
+		)
+	else()
+		add_library( magma ${libmagma_all} )
+		target_link_libraries( magma
+			${LAPACK_LIBRARIES}
+		)
+	endif()
 endif()
 
 
@@ -371,14 +378,22 @@
 include_directories( sparse-iter/include )
 include_directories( sparse-iter/control )
 include_directories( testing )
-cuda_add_library( magma_sparse ${libsparse_all} )
-target_link_libraries( magma_sparse
-	magma
-	${LAPACK_LIBRARIES}
-	${CUDA_CUDART_LIBRARY}
-	${CUDA_CUBLAS_LIBRARIES}
-	${CUDA_cusparse_LIBRARY}
-)
+if ( CUDA_FOUND )
+	cuda_add_library( magma_sparse ${libsparse_all} )
+	target_link_libraries( magma_sparse
+		magma
+		${LAPACK_LIBRARIES}
+		${CUDA_CUDART_LIBRARY}
+		${CUDA_CUBLAS_LIBRARIES}
+		${CUDA_cusparse_LIBRARY}
+	)
+else()
+	add_library( magma_sparse ${libsparse_all} )
+	target_link_libraries( magma_sparse
+		magma
+		${LAPACK_LIBRARIES}
+	)
+endif()
 set( LIBS_SPARSE ${LIBS} magma_sparse )
 
 
